docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,219 @@
+
+For example, this could allow users to create clusters using tooling such as Terraform or Kops and allow them to add CAPI Machine infrastructure as a day 2 operation.
+
+This will also ease adoption of CAPI in heterogeneous real world environments with restricted privileges and where the provider infrastructure for the cluster needs to be managed out of band.
+
+### Goals
+
+- Introduce support for "externally managed" infrastructure consistently across CAPI providers.
+- The machine controller must be able to operate and manage machines when the infastructure is "externally managed".
+- Reuse existing InfraCluster CRDs in "externally managed" clusters to minimise differences between the two topologies.
+
+### Non-Goals/Future Work
+
+- Modify existing managed behaviour.
+- Automatically mark InfraCluster resources as ready (this will be up to the external management component initially).
+
+## Proposal
+
+A new contract will be defined around an annotation `cluster.x-k8s.io/managed-by: "<name-of-system>"`.
+This annotation will identify a system which is providing external management of the infrastructure.
+
+It is expected that all providers will adhere to the contract being defined by this proposal.
+
+When this annotation is present on an InfraCluster resource, the InfraCluster controller is expected to ignore the resource and not perform any reconciliation.
+Importantly, it will not modify the resource or its status in any way.
+
+Additionally, the external management system must provide all required fields within the spec of the InfraCluster and must adhere to the CAPI provider contract and set the InfraCluster status to be ready when it is appropriate to do so.
+
+While an "externally managed" InfraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, CAPI will still be able to create compute nodes within it.
+
+The machine controller must be able to operate without hard dependencies regardless of the cluster infrastructure being managed or externally managed.
+![](https://i.imgur.com/nA61XJt.png)
+
+### User Stories
+
+#### Story 1 - Alternate control plane provisioning with user managed infrastructure
+As a cluster provider I want to use CAPI in my service offering to orchestrate Kubernetes bootstraping while letting workload cluster operators own their infrastructure lifecycle.
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstraping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
+
+#### Story 3 - Consuming existing cloud infrastructure
+As a cluster operator I want to use CAPI to orchestate Kubernetes bootstraping while reusing infrastructure that has already been created in the organisation either by me or another team.
+
+### Implementation Details/Notes/Constraints
+
+**Managed**
```

Comment:
What about adding a note that reminds that Mananged = InfraCluster without  the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation